ggircs-app/app/components/XmlDiff/RenderDiff.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState, useEffect } from "react";
+import { diffLines, formatLines } from "unidiff";
+import { parseDiff, Diff, Hunk, tokenize } from "react-diff-view";
+import "react-diff-view/style/index.css";
+import LoadingSpinner from "components/LoadingSpinner";
+
+const EMPTY_HUNKS = [];
+
+interface Props {
+  oldText: string;
+  newText: string;
+  collapse: boolean;
+}
+
+export const RenderDiff: React.FunctionComponent<Props> = ({
+  oldText,
+  newText,
+  collapse,
+}) => {
+  const [loaded, setLoaded] = useState(false);
+
+  useEffect(() => {
+    setLoaded(true);
+  }, []);
```

Comment:
Why is this needed? Setting it immediately to `true` is a bit confusing. Is there a callback somewhere where `setLoaded(true)` would be more appropriate?

Reply:
This is supposed to be like `ComponentDidMount`. When the page loads quickly this gets immediately set to true, but when it's taking a couple seconds to load (ie: comparing large diffs) I'm using this `loaded` state variable & useEffect to render the loading spinner until the whole component has loaded. I could rename the state variable to `isLoading` instead if that's more clear?

Reply:
I see. So this is simply to first render a loading spinner while the full, synchronous render is computing? This is not ideal, and we'd probably want to use a webworker for the diff calculation to keep the UI responsive, but that will do for now. Can you add a comment here to explain what this is for?